### DISCUSSION TITLE
Noob question - How to encode custom phoneme data

### DISCUSSION THREAD

**Discussion by kafan1986** [coqui-ai/TTS#1482] [General]
I have custom text data represented in phoneme (pronunciation) format. Now there are two additional data points, one is syllable boundaries and other is stress (if present of or not a particular syllable. Wanted to understand how should I encode this data? For example my phoneme for a word is abc."def.ghi where a,b,c,d,e,f,g,h,i individually represent my phonetic (sound) units, . represents syllable boundaries and " represent the presence of stress at the syllable. I can encode phonetic units easily, also syllable boundaries are not a big concern, as I believe it can be treated as just another punctuation character. Now question is, how should I address the stress information, as " which represents stress is applicable on the overall syllable (in the previous example "def")?

Hope I was able to explain my problem.

**Comment by erogol** [comment]
Phoneme symbols include characters to indicate stress. You can follow the same conversion and let the model learn a more expressive speech. 

Then at inference, you can type the phonemes as you want. However, our current inference endpoint does not support phoneme injection to the input text. You need to do it manually for now. The good news, it is WIP
